R/readutils.R

```diff
@@ -343,7 +343,40 @@ readtextcf <- function(file, T=48, sym=TRUE, path="", skip=1, check.t=0, ind.vec
 
   return (invisible(ret))
 }
+#' @title reading reweighting factors for a list of gauge configuration 
+#'        and random samples from ASCII files in CMI format
+#' @param file_names_to_read list of filenames for the reweighting factors
+#' @param gauge_conf_list <- a list of integers with the indices of the gauge configs
+#' @param nsamples number of stochastic samples used for computing the reweighting factors
+readcmi.rw <- function( file_names_to_read, gauge_conf_list, nsamples, monomial_id ) 
+{
+  stopifnot(length(gauge_conf_list)==length(file_names_to_read)) 
+  ret <- rw_meta(conf.index=gauge_conf_list)
+  tmp <- readcmidatafiles(files=file_names_to_read,skip=0,verbose=TRUE,colClasses=c("integer","integer","numeric","numeric","numeric","numeric","numeric"))
+
+# Select the reweighting factor for a particular monomial
+  
+  tmp <- filter(tmp,V1==monomial_id)
```

Comment:
```filter``` is not base R, if you need to use it, specify the dependency and use the namespace when calling the function

Reply:
solved

Reply:
you need to require the namespace in this function (and fail gracefully if it's missing as dplyr should be an optional dependency) and use the namespace when calling functions from dplyr

Reply:
see uses of external functions elsewhere and in particular those cases using optional dependencies

Reply:
Now I use:
```
dplyr_avail <- requireNamespace("dplyr")
stopifnot(dplyr_avail)
tmp <- dplyr::filter(tmp,V1==monomial_id)

```